File: csharp/rotational-cipher/RotationalCipher.cs
using System;
using System.Linq;

public static class RotationalCipher
{
    public static string Rotate(string text, int shiftKey) => text.Aggregate("", (current, c) => current + Rotate(c, shiftKey));

    private static char Rotate(char character, int shiftKey)
    {
        if (character >= 'a' && character <= 'z')
        {
            return (char)(((character - 'a' + shiftKey) % 26) + 'a');
        }
        else if (character >= 'A' && character <= 'Z')
        {
            return (char)(((character - 'A' + shiftKey) % 26) + 'A');
        }

        return character;
    }
}

File: cpp/rotational-cipher/rotational_cipher.cpp
#include "rotational_cipher.h"

namespace rotational_cipher
{

    char rotate(const char c, int shift)
    {
        if (c >= 'a' && c <= 'z')
        {
            return (char)(((c - 'a' + shift) % 26) + 'a');
        }
        else if (c >= 'A' && c <= 'Z')
        {
            return (char)(((c - 'A' + shift) % 26) + 'A');
        }

        return c;
    }

    std::string rotate(const std::string &text, int shift)
    {
        std::string result;

        for (auto i : text)
        {
            result += rotate(i, shift);
        }

        return result;
    }

} // namespace rotational_cipher
